### PR TITLE
Add support for Vaillant ecoTEC plus VU INT IV 246/5-5 H (0010021961)

### PR DESCRIPTION
**Pull request by andr2000**
With this binding most of the commands of the Vaillant ecoTEC
plus VU INT IV 246/5-5 H (0010021961) are read correctly.

Checked with "MF=Vaillant;ID=BAI00;SW=0104;HW=7803",
loaded "vaillant/bai.0010015600.inc" ([PROD='0010021961']).

Signed-off-by: Oleksandr Andrushchenko <andr2000@gmail.com>

### CODE REVIEW

**Comment by andr2000** [comment]
@cameleon77 and @andig migth be interested as well.
There are at least some of the commands are not present on my system:
```Status16 ERR: invalid position in decode
Status ERR: invalid position in decode```

**Comment by andig** [comment]
@andr2000 seems you've added your prodid twice to the PR? 

As for `PartloadHcKW`: the name is misleading- this is the MAX power the BAI might use (d.00).
`ModulationTempDesired`: also misleading. This is the percentage of the NOMINAL power that the BAI has that IS being used.

Example for my BAI:
- `PartloadHcKW` 6: 6kW max power out of 14.6kW
- `ModulationTempDesired` 19: 19% out of 14.6 is 2.77kW which is the minimal power

**Comment by andr2000** [comment]
> @andr2000 seems you've added your prodid twice to the PR?
I added it to en and de files, I think those need to be in sync?
ebusd-2.1.x/de/vaillant/08.bai.csv
ebusd-2.1.x/en/vaillant/08.bai.csv

File: ebusd-2.1.x/de/vaillant/08.bai.csv [hunk 1]
```diff
@@ -13,7 +13,7 @@
 [PROD='0010006341']!load,bai.0010006341.inc,,,
 [PROD='0010007508';'0010007510';'0010007512';'0010007514';'0010007516';'0010007518';'0010007520';'0010007522';'0010007524';'0010007526';'0010007688';'0010007692';'0010007696';'0010007700';'0010007704']!load,bai.0010007508.inc,,,
 [PROD='0010010674';'0010010676';'0010010678';'207']!load,bai.0010010674.inc,,,
-[PROD='0010014629';'0010014630';'0010014631';'0010014632';'0010014633';'0010015596';'0010015597';'0010015598';'0010015599';'0010015600';'0010015601';'0010015602';'0010015603';'0010015604';'0010015605';'0010015606';'0010015607';'0010015608';'0010015609';'0010015610';'0010015611';'0010015612';'0010015613';'0010015614';'0010019268';'0010021888']!load,bai.0010015600.inc,,,
+[PROD='0010014629';'0010014630';'0010014631';'0010014632';'0010014633';'0010015596';'0010015597';'0010015598';'0010015599';'0010015600';'0010015601';'0010015602';'0010015603';'0010015604';'0010015605';'0010015606';'0010015607';'0010015608';'0010015609';'0010015610';'0010015611';'0010015612';'0010015613';'0010015614';'0010019268';'0010021888';'0010021961';'0010021961']!load,bai.0010015600.inc,,,
```
Comment (john30) [inline ebusd-2.1.x/de/vaillant/08.bai.csv]: don't add the same product ID twice

File: ebusd-2.1.x/en/vaillant/08.bai.csv [hunk 2]
```diff
@@ -13,7 +13,7 @@
 [PROD='0010006341']!load,bai.0010006341.inc,,,
 [PROD='0010007508';'0010007510';'0010007512';'0010007514';'0010007516';'0010007518';'0010007520';'0010007522';'0010007524';'0010007526';'0010007688';'0010007692';'0010007696';'0010007700';'0010007704']!load,bai.0010007508.inc,,,
 [PROD='0010010674';'0010010676';'0010010678';'207']!load,bai.0010010674.inc,,,
-[PROD='0010014629';'0010014630';'0010014631';'0010014632';'0010014633';'0010015596';'0010015597';'0010015598';'0010015599';'0010015600';'0010015601';'0010015602';'0010015603';'0010015604';'0010015605';'0010015606';'0010015607';'0010015608';'0010015609';'0010015610';'0010015611';'0010015612';'0010015613';'0010015614';'0010019268';'0010021888']!load,bai.0010015600.inc,,,
+[PROD='0010014629';'0010014630';'0010014631';'0010014632';'0010014633';'0010015596';'0010015597';'0010015598';'0010015599';'0010015600';'0010015601';'0010015602';'0010015603';'0010015604';'0010015605';'0010015606';'0010015607';'0010015608';'0010015609';'0010015610';'0010015611';'0010015612';'0010015613';'0010015614';'0010019268';'0010021888';'0010021961';'0010021961']!load,bai.0010015600.inc,,,
```
Comment (john30) [inline ebusd-2.1.x/en/vaillant/08.bai.csv]: don't add the same product ID twice

Comment (andr2000) [reply ebusd-2.1.x/en/vaillant/08.bai.csv]: Ah, now I see. Sorry about that

**Comment by andr2000** [comment]
Fixed now

**Comment by andr2000** [comment]
Please hold on, I'd like to make an extra check here

**Comment by andr2000** [comment]
@john30 the reason I asked to hold on is that I checked with installation manual for the boiler (appendix with d.xx codes) and tried this pull request: I see that old behavior, when I used fall-back to bai.308523.inc, produces many more commands which seem to be in sync with the manual, but some of them produce wrong result, e.g. 
```
ebusctl r -f -V PartloadHcKW
ERR: invalid position in decode
```
If I use the changes from this pull request (and effectively map my device to use bai.0010015600.inc) I can read those values w/o issues, but the list of commands is now not full.
Could you please suggest the way to go? What is the right way to get my boiler supported?
I can see couple of ways to do so:
1. Create a new bai.0010021961.inc and copy-paste all the supported commands from other bai files with possible fixes/tunning
2. Extend bai.0010015600.inc by adding a new line with support for my device and include an extra bai.0010021961.inc which will add missed commands
3. Anything else I can try?

**Comment by john30** [comment]
the best is probably to create a new inc file for your HW

**Comment by andr2000** [comment]
@john30 could you please review? It only has EN version for now, but I heard that you are working on having messages and translations separate, so not sure what needs to be done wrt DE version here.
What is more the boilers in question are sold in Ukraine and EN would be more appropriate here IMO
Thank you

**Comment by andr2000** [comment]
I'm seeing some issues though:

**Comment by andig** [comment]
Could you add the DE version, too? Right now all files are in sync between the languages.

**Comment by andr2000** [comment]
The only reason I didn't add DE is I don't speak German... So, I can try adding, but I cannot tell if the wording exactly matches parameter meaning...
@john30 - what is your preference here?

**Comment by john30** [comment]
I'd also prefer having a german version as well for the time being. Just copy and paste from the current german files.

**Comment by andr2000** [comment]
@john30  Sorry for being so clumsy.... I have finally added German translation, unfortunately I cannot check it as I don't speak German. Also some of the parameters are not translated in German files as well, so this is what I won't be able to do....

**Comment by andr2000** [comment]
@john30 I hope I have addressed all the comments now. Could you please review?

**Comment by andr2000** [comment]
gentle reminder....

File: ebusd-2.1.x/de/vaillant/08.bai.csv [hunk 3]
```diff
@@ -16,6 +16,7 @@
 [PROD='0010014629';'0010014630';'0010014631';'0010014632';'0010014633';'0010015596';'0010015597';'0010015598';'0010015599';'0010015600';'0010015601';'0010015602';'0010015603';'0010015604';'0010015605';'0010015606';'0010015607';'0010015608';'0010015609';'0010015610';'0010015611';'0010015612';'0010015613';'0010015614';'0010019268';'0010021888']!load,bai.0010015600.inc,,,
 [PROD='0020066007';'187']!load,bai.0020066007.inc,,,
 [PROD='0010004276';'0010004277';'0010004279';'0010004280';'0010004281';'0010004282';'0010004283';'0010004285';'0010004286';'0010004288';'0010004289';'0010004290';'0010004291';'0010004292';'0010004336';'0010004337';'0010004338';'0010004339';'0010004340';'0010005466';'0010005467';'0010005468';'0010005469';'0010010392';'0010010393';'0010010394';'0010010400';'0020051714';'0020051715';'0020051716';'0020051717';'174']!load,bai.308523.inc,,,
+[PROD='0010021961';'0010021962';'0010021963';'0010021964';'0010021965';'0010021966';'0010021967']!load,bai.0010021961.inc,,,
```
Comment (john30) [inline ebusd-2.1.x/de/vaillant/08.bai.csv]: where did you get the list of applicable product codes from?

**Comment by john30** [comment]
did you verify all message definitions work as expected?

File: ebusd-2.1.x/de/vaillant/08.bai.csv (hunk 3 above)
Comment (andr2000) [reply ebusd-2.1.x/de/vaillant/08.bai.csv]: from the hard copy of the manual

**Comment by andr2000** [comment]
Yes, I have verified and can confirm that it works.

**Comment by john30** [comment]
bad: the IDs of the files are not the same.
Which is correct for FlowsetHwcMax: 0f04 or a600?

**Comment by andr2000** [comment]
My bad, sorry. I never tried to **write**, only read:
```
~# ebusctl f -f -a FlowsetHwcMax
r,bai,FlowsetHwcMax,d.78 Max value flow temp storage,,08,b509,0da600,temp,s,D2C,,°C,storage charging temperature limit (target flow temperature in storage mode)
w,bai,FlowsetHwcMax,d.78 Max value flow temp storage,,08,b509,0ea600,temp,m,D2C,,°C,storage charging temperature limit (target flow temperature in storage mode)

root@vaillant:~# ebusctl r -f FlowsetHwcMax
75.00
```

**Comment by john30** [comment]
ok, I've corrected that in e539ff5 alreay

**Comment by andr2000** [comment]
thank you!

**Pull request merged**